guides/hosting/installation-updates/docker.md

```diff
@@ -0,0 +1,272 @@
+| FPM_PM                               | dynamic          | [See PHP FPM documentation](https://www.php.net/manual/en/install.fpm.configuration.php) |
+| FPM_PM_MAX_CHILDREN                  | 5                | [See PHP FPM documentation](https://www.php.net/manual/en/install.fpm.configuration.php) |
+| FPM_PM_START_SERVERS                 | 2                | [See PHP FPM documentation](https://www.php.net/manual/en/install.fpm.configuration.php) |
+| FPM_PM_MIN_SPARE_SERVERS             | 1                | [See PHP FPM documentation](https://www.php.net/manual/en/install.fpm.configuration.php) |
+| FPM_PM_MAX_SPARE_SERVERS             | 3                | [See PHP FPM documentation](https://www.php.net/manual/en/install.fpm.configuration.php) |
+
+## Possible Mounts
+
+::: info
+Our recommandation is to store all files in an external storage provider to not mount any volumes. Refer to [official Shopware docs for setup](https://developer.shopware.com/docs/guides/hosting/infrastructure/filesystem).
+:::
+
+In a very basic setup when all files are stored locally you need 5 volumes:
+
+| Usage                  | Path                           |
+|------------------------|--------------------------------|
+| invoices/private files | /var/www/html/files            |
+| theme files            | /var/www/html/public/theme     |
+| images                 | /var/www/html/public/media     |
+| image thumbnails       | /var/www/html/public/thumbnail |
+| generated sitemap      | /var/www/html/public/sitemap   |
+
+
+Shopware logs by default to `var/log`, but when `shopware/docker` Composer package is installed, we change it to stdout. This means you can use `docker logs` to see the logs or use logging driver to forward the logs to a logging service.
+
+## Ideal Setup
+
+The ideal setup requires an external storage provider like S3. In that way you can don't need any mounts and can scale the instances without any problems.
+
+Additionally Redis is required for the session storage and the cache, so the Browser sessions are shared between all instances and cache invalidations are happening on all instances.
+
+## Typical Setup
+
+The docker image starts in entrypoint PHP-FPM / Caddy. So you will need to start a extra container to run maintenance tasks like to install Shopware, install plugins, or run the update. This can be done by installing the [Deployment Helper](./deployments/deployment-helper.md) and creating one container and running as entrypoint `/setup`
+
+Here is an example of the `compose.yml` (`docker-compose.yml`) with all the requierd services:
+
+```yaml
+x-environment: &shopware
+  image: local
+  build:
+    context: .
+  environment:
+    APP_ENV: prod
+    DATABASE_URL: 'mysql://shopware:shopware@database/shopware'
+    DATABASE_HOST: 'database'
+    APP_URL: 'http://localhost:8000'
+    APP_SECRET: 'test'
+    PHP_SESSION_HANDLER: redis
+    PHP_SESSION_SAVE_PATH: 'tcp://cache:6379/1'
```

Comment:
If we provided the database service (which is far from being production-ready or highly available), shouldn't we also provide example for the cache service (Redis)? Maybe define both in a separate stack and add network config to the `x-environment: $shopware`? This does make the config more complicated, but also draws the line between shopware- and external-provided services, makes it easier to run multiple web/app stacks with one shared stack running independent services (maria, redis, adminer, ...).

Reply:
I wanted to visualize how a deployment could look like and how everything works with each other. for production I would just document later on a new page the kubernetes operator https://github.com/shopware/shopware-operator

Reply:
Would you stay on a docker compose, or I draw something in draw.io 🤔

Reply:
I like the idea to have a working running docker-compose.yaml file, so I would stick with it. A side node that other services like valkey could also be used, is maybe also worth it.